src/e2e/java/teammates/e2e/cases/lnp/StudentUpdateLNPTest.java

```diff
@@ -0,0 +1,331 @@
+                        .build();
+
+                feedbackSessions.put(FEEDBACK_SESSION_NAME, session);
+
+                return feedbackSessions;
+            }
+
+            @Override
+            protected Map<String, FeedbackQuestionAttributes> generateFeedbackQuestions() {
+                ArrayList<FeedbackParticipantType> showResponses = new ArrayList<>();
+                showResponses.add(FeedbackParticipantType.RECEIVER);
+                showResponses.add(FeedbackParticipantType.INSTRUCTORS);
+
+                ArrayList<FeedbackParticipantType> showGiverName = new ArrayList<>();
+                showGiverName.add(FeedbackParticipantType.INSTRUCTORS);
+
+                ArrayList<FeedbackParticipantType> showRecepientName = new ArrayList<>();
+                showRecepientName.add(FeedbackParticipantType.INSTRUCTORS);
+
+                Map<String, FeedbackQuestionAttributes> feedbackQuestions = new LinkedHashMap<>();
+                FeedbackQuestionDetails details = new FeedbackTextQuestionDetails(QUESTION_TEXT);
+
+                feedbackQuestions.put(QUESTION_ID,
+                        FeedbackQuestionAttributes.builder()
+                                .withFeedbackSessionName(FEEDBACK_SESSION_NAME)
+                                .withQuestionDescription(QUESTION_TEXT)
+                                .withCourseId(COURSE_ID)
+                                .withQuestionDetails(details)
+                                .withQuestionNumber(1)
+                                .withGiverType(FeedbackParticipantType.STUDENTS)
+                                .withRecipientType(FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF)
+                                .withShowResponsesTo(showResponses)
+                                .withShowGiverNameTo(showGiverName)
+                                .withShowRecipientNameTo(showRecepientName)
+                                .build()
+                );
+                return feedbackQuestions;
+            }
+
+            @Override
+            protected Map<String, FeedbackResponseAttributes> generateFeedbackResponses() {
+                Map<String, FeedbackResponseAttributes> feedbackResponses = new HashMap<>();
+
+                for (int i = 1; i <= NUMBER_OF_FEEDBACK_RESPONSES; i++) {
+                    String responseText = FEEDBACK_RESPONSE_ID + " " + i;
+                    FeedbackTextResponseDetails details =
+                            new FeedbackTextResponseDetails(responseText);
+
+                    feedbackResponses.put(responseText,
+                            FeedbackResponseAttributes.builder(QUESTION_ID,
+                                STUDENT_EMAIL,
+                                STUDENT_EMAIL)
+                                .withCourseId(COURSE_ID)
+                                .withFeedbackSessionName(FEEDBACK_SESSION_NAME)
+                                .withGiverSection(GIVER_SECTION_NAME)
```

Comment:
I think update this field will also cause cascade update. Have u considered this?

Reply:
Yup, I am considering the worst case of a general student update here. 

Maybe in the future, if we want a more fine-grained LNP test, we can separate it out to 2 test cases, one for course_id, and one for section name.

Reply:
> Maybe in the future, if we want a more fine-grained LNP test, we can separate it out to 2 test cases, one for course_id, and one for section name.

I think this is more important to do. See if you can finish before your internship ends :P

Reply:
If you check the code, there are three different conditions for student cascade update:
- Changing student email
- Changing student team
- Changing student section

For LnP test to be successful in catching bottleneck operations, they need to be separate.